src/negotiation/order.ts

```diff
@@ -16,8 +16,125 @@ export interface OrderAsset {
   nominalAmount: string;
 }
 
+export interface TakerCriteria {
+  buy: TakerCriteriaAsset;
+  sell: TakerCriteriaAsset;
+  minRate?: number;
+}
+
+export interface TakerCriteriaAsset {
+  ledger: string;
+  asset: string;
+  minNominalAmount?: string;
+  maxNominalAmount?: string;
+}
+
+/**
+ * @description An order handler for the taker. It has helper functions to facilitate the handler of an
+ * order by a taker. This should only be created via `TakerNegotiatior.getOrder()` and should not be constructed from
+ * scratch.
+ * @param orderParams - The raw parameters of the order.
+ * @param criteria - The criteria used to get this order.
+ * @param takeOrder - Function passed from the `TakerNegotiator` to coordinate calls to `cnd` and the maker to effectively
+ * take the order and start the atomic swap execution.
+ */
+export class Order {
+  /**
+   * @description **Note: This should not be used, `Order` should be created by using `TakerNegotiatior.getOrder()`
+   * @param orderParams - The parameters of the order, as received from the maker.
+   * @param criteria - The criteria used to filter/retrieve this order.
+   * @param takeOrder - `TakerNegotiator.execAndTakeOrder()`
+   */
+  constructor(
+    public readonly orderParams: OrderParams,
+    public readonly criteria: TakerCriteria,
+    public readonly takeOrder: (
+      orderParams: OrderParams
+    ) => Promise<Swap | undefined>
+  ) {}
+
+  /**
+   * @description: Return whether an order matches the passed criteria.
+   */
+  public matches(): boolean {
+    return (
+      assetMatches(this.criteria.buy, this.orderParams.bid) &&
+      assetMatches(this.criteria.sell, this.orderParams.ask) &&
+      rateMatches(this.criteria, this.orderParams)
+    );
+  }
+
+  /**
+   * @description Check that the order is valid and safe. Ensure that all properties are set and that the expiries
+   * are relatively safe. It does not check whether the ledgers/assets are correct, this is done with `Order.matches()`.
+   */
+  public isValid(): boolean {
+    return areOrderParamsValid(this.orderParams);
+  }
+
+  /**
+   * @description Initiate the swap execution and tells the maker that we are taking this order.
```

Comment:
```suggestion
   * @description Initiates the swap execution and tells the maker that we are taking this order.
```

or remove the `s` on tells